saleor/core/payments.py

```diff
@@ -0,0 +1,76 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import TYPE_CHECKING, List
+
+if TYPE_CHECKING:
+    from saleor.payment.interface import PaymentData, GatewayResponse, CustomerSource
+
+
+class Gateway(Enum):
+    """Possible gateway values.
+
+    TODO: Create this in runtime based on available plugins
+    """
+
+    DUMMY = "dummy"
+    BRAINTREE = "braintree"
+    RAZORPAY = "razorpay"
+    STRIPE = "stripe"
+
+
+class PaymentInterface(ABC):
+    @abstractmethod
+    def list_payment_gateways(self) -> List[Gateway]:
```

Comment:
It would be nice to add some docstrings about the purpose of each function. Not sure if we should have it here or in the extensions manager?

Reply:
I would put docstring in Gateway since it is entry point for clients code.
Additionally some explanation might be put into interface.